### PR TITLE
Interaction ols

### PR DESCRIPTION
**Pull request by daslu**
Converting the [Interaction-ols](https://github.com/scicloj/scicloj.ml-tutorials/blob/main/src/scicloj/ml/interactions_ols.clj) tutorial from the scicloj.ml tutorials:
* not using scicloj.ml, but rather the internal libraries that it wraps
* writing it as a Kindly notebook
* reading the data at runtime from a separate repo [datarium-CSV](https://github.com/scicloj/datarium-CSV)

### CODE REVIEW

**Comment by behrica** [comment]
I my latest docs:
https://scicloj.github.io/noj/

I am proposing to only use the pipeline style , when needed.
https://scicloj.github.io/noj/automl.html#best-practices-for-data-transformation-steps-in-or

I would even go a step further, by saying, we should only use the pipelines when doing AutoML, so when using function `ml/evaluate-pipelines`

So for most simple tutorials, which show "one model", with "one train / predict pair" I would suggest to use the simple functions
`ml/train` and `ml/predict`
This would change most lines of the tutorial, nut would be easy to do.
I suggest you to do this, at least for educational purpose.

File: notebooks/interactions_ols.clj [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  (mm/pipeline
+   (tcpipe/drop-columns [:newspaper])
+   (tcpipe/add-column :youtube*facebook (fn [ds] (fun/* (ds :youtube) (ds :facebook))))
+   (tmd.mm/set-inference-target :sales)
+   {:metamorph/id :model}(mm.ml/model {:model-type :smile.regression/ordinary-least-square})))
+
+(md "Again we evaluate the model,")
+(def evaluations
+  (mm.ml/evaluate-pipelines
+   [pipe-interaction]
```
Comment (behrica) [inline notebooks/interactions_ols.clj:74]: You have a single pipe-fn, his means using the pipeline concept has no advantage.
Specially in this case of "regression", we would likely never do AutoML.

Very often in regression, we don't want to find simply teh "Best model", bu we want to learn something about the underlying data / phenomena.

**Comment by daslu** [comment]
Zulip discussion:
https://clojurians.zulipchat.com/#narrow/stream/321125-noj-dev/topic/ols.20interaction.20tutorial

File: notebooks/interactions_ols.clj (hunk 1 above)
Comment (behrica) [reply notebooks/interactions_ols.clj:74]: But your comment regarding test / train split is correct as well.
The syntax with `ml/evaluate` and pipelines is a bit more compact and 
avoids  some `def`s

File: notebooks/interactions_ols.clj [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   (tc/split->seq marketing :holdout)
+   loss/rmse
+   :loss
+   {:other-metrices [{:name :r2
```
Comment (behrica) [inline notebooks/interactions_ols.clj:78]: this is as well more compact with pipelines

Comment (behrica) [reply notebooks/interactions_ols.clj:78]: So overall it's a closed call, if  pipelines or a train/predict pair are better  for this example.
We can as well leave it as is.

Comment (daslu) [reply notebooks/interactions_ols.clj:78]: Thanks.

Then, at least, I will extract the fixed parts from the pipelines, as you suggested in the Zulip chat.

Comment (daslu) [reply notebooks/interactions_ols.clj:78]: @behrica I moved some of the fixed parts to the preprocessing (outside of pipelines).

Adding the `youtube*facebook` column is still in the pipeline, as it makes it convenient to run different pipelines over the same datasets.

What do yu think?

**Comment by behrica** [comment]
Looks good. We can merge it.

**Comment by daslu** [comment]
Thanks for your patient guidance, @behrica. I am learning a lot from this.

**Pull request merged**